src/coreclr/jit/assertionprop.cpp

```diff
@@ -3910,7 +3853,7 @@ GenTree* Compiler::optAssertionProp_ModDiv(ASSERT_VALARG_TP assertions, GenTreeO
 
         // OAK_[NOT]_EQUAL assertion with op1 being O1K_CONSTANT_LOOP_BND
         // representing "(X relop CNS) ==/!= 0" assertion.
-        if (!curAssertion->IsConstantBound())
+        if (!curAssertion->IsConstantBound() && !curAssertion->IsConstantBoundUnsigned())
```

Comment:
Here for `optAssertionProp_ModDiv` I added `IsConstantBoundUnsigned` (unsigned comparisons against constants).

Previously we could optimize div only for `x > cns` assert. This adds `(uint)x < cns` case as well.